src/systemd/cockpit.service.in

```diff
@@ -3,14 +3,16 @@ Description=Cockpit Web Service
 Documentation=man:cockpit-ws(8)
 Requires=cockpit.socket
 Requires=cockpit-wsinstance-http.socket cockpit-wsinstance-https-factory.socket
-After=cockpit-wsinstance-http.socket cockpit-wsinstance-https-factory.socket
+# we need to start before the sockets so that the dynamic user exists
+Before=cockpit-wsinstance-http.socket cockpit-wsinstance-https-factory.socket
```

Comment:
I feel like there's a race here...

Reply:
Specifically: if `cockpit.service` manages to come online (ie: by socket activation the first time someone connects) and receives the first few bytes of data, noticing that it's not a TLS connection, it will immediately attempt to contact the `http` socket, which according to the dependency information here is not necessarily there yet.

I think that this race is *probably* not going to be lost often (since spawning the process takes time, and systemd is probably directly on its way to create those listeners immediate after the process is forked off) but it's definitely a possibility...

Reply:
The way I dealt with this in #16811 is by having a separate `cockpit-wsinstance-socket-owner.service` which was the thing that created the dynamic user.  Everything depended on that.

Reply:
Thanks for spotting! I agree, this is racy. I adopted your approach with the intermediate unit. Tested with static and with dynamic user, works nicely!